smartsim/_core/generation/operations.py

```diff
@@ -0,0 +1,298 @@
+import base64
+import os
+import pathlib
+import pickle
+import sys
+import typing as t
+from dataclasses import dataclass, field
+
+from ..commands import Command
+
+entry_point_path = "smartsim._core.entrypoints.file_operations"
+"""Path to file operations module"""
+
+copy_cmd = "copy"
+"""Copy file operations command"""
+symlink_cmd = "symlink"
+"""Symlink file operations command"""
+configure_cmd = "configure"
+"""Configure file operations command"""
+
+
+def create_final_dest(
+    job_root_path: pathlib.Path, dest: t.Union[pathlib.Path, None]
+) -> str:
+    """Combine the job root path and destination path. Return as a string for
+    entry point consumption.
+
+    :param job_root_path: Job root path
+    :param dest: Destination path
+    :return: Combined path
+    :raises ValueError: An error occurred during path combination
+    """
+    try:
+        combined_path = job_root_path
+        if dest:
+            combined_path = job_root_path / dest
+        return str(combined_path)
+    except Exception as e:
+        raise ValueError(f"Error combining paths: {e}")
+
+
+def check_src_and_dest_path(
+    src: pathlib.Path, dest: t.Union[pathlib.Path, None]
+) -> None:
+    """Validate that the provided source and destination paths are
+    of type pathlib.Path
+
+    :param src: The source path to be checked.
+    :param dest: The destination path to be checked.
+    :raises TypeError: If either src or dest is not an instance of pathlib.Path
+    """
+    if not isinstance(src, pathlib.Path):
+        raise TypeError(f"src must be of type pathlib.Path, not {type(src).__name__}")
+    if dest is not None and not isinstance(dest, pathlib.Path):
+        raise TypeError(
+            f"dest must be of type pathlib.Path or None, not {type(dest).__name__}"
+        )
+    if isinstance(dest, pathlib.Path) and not dest.is_absolute():
+        raise ValueError("Invalid destination path")
+    if isinstance(dest, pathlib.Path) and " " in str(dest):
+        raise ValueError("Path contains spaces, which are not allowed")
+
+    # TODO I want to add the check below but I do not think this works for remote jobs
+    # full_path = path.abspath(file_path)
+    #     if path.isfile(full_path):
+    #         return full_path
+    #     if path.isdir(full_path):
+    #         return full_path
+
+
+def check_run_path(run_path: pathlib.Path) -> None:
```

Comment:
Similar "can we mark this protected" question here
```suggestion
def _check_run_path(run_path: pathlib.Path) -> None:
```